app/src/androidTest/java/org/simple/clinic/protocolv2/ProtocolRepositoryAndroidTest.kt

```diff
@@ -0,0 +1,162 @@
+  lateinit var testData: TestData
+
+  @get:Rule
+  val authenticationRule = AuthenticationRule()
+
+  @Before
+  fun setUp() {
+    TestClinicApp.appComponent().inject(this)
+  }
+
+  // TODO: Remove user auth when this test is removed.
+  // TODO: Make protocolUuid non-null in both Facility and FacilityPayload.
+  @Test
+  fun protocol_ID_should_not_be_empty_when_feature_is_enabled() {
+    val config = configProvider.blockingGet()
+    if (config.isProtocolDrugSyncEnabled) {
+      val currentUser = userSession.loggedInUserImmediate()!!
+      val currentFacility = facilityRepository.currentFacility(currentUser).blockingFirst()
+      assertThat(currentFacility.protocolUuid).isNotNull()
+    }
+  }
+
+  // TODO: Remove user auth when this test is removed.
+  @Test
+  fun when_feature_is_disabled_then_default_drugs_should_be_returned() {
+    val config = configProvider.blockingGet()
+    if (config.isProtocolDrugSyncEnabled.not()) {
+      val randomProtocolId = UUID.randomUUID()
+      val drugs = protocolRepository.drugsForProtocol(randomProtocolId).blockingFirst()
+      assertThat(drugs).isEqualTo(protocolRepository.defaultProtocolDrugs())
+    }
+  }
+
+  @Test
+  fun when_protocols_are_not_present_in_database_then_default_drugs_should_be_returned() {
+    database.clearAllTables()
+
+    val randomProtocolId = UUID.randomUUID()
+    val drugs = protocolRepository.drugsForProtocol(randomProtocolId).blockingFirst()
+    assertThat(drugs).isEqualTo(protocolRepository.defaultProtocolDrugs())
+  }
+
+  @Test
+  fun drugs_are_present_but_feature_is_disabled_then_default_drugs_should_be_returned() {
+    val config = configProvider.blockingGet()
+    if (config.isProtocolDrugSyncEnabled.not()) {
+      val protocol1 = testData.protocol()
+      database.protocolDao().save(listOf(protocol1))
+
+      val drug1 = testData.protocolDrug(protocolUuid = protocol1.uuid)
+      database.protocolDrugDao().save(listOf(drug1))
+
+      val drugs = protocolRepository.drugsForProtocol(protocol1.uuid).blockingFirst()
+      assertThat(drugs).isEqualTo(protocolRepository.defaultProtocolDrugs())
+    }
+  }
+
+  @Test
+  fun when_drugs_are_not_present_for_a_specific_protocol_then_default_values_should_be_returned() {
+    database.clearAllTables()
+
+    val protocol1 = testData.protocol()
+    val protocol2 = testData.protocol()
+    database.protocolDao().save(listOf(protocol1, protocol2))
+
+    val drug1 = testData.protocolDrug(protocolUuid = protocol1.uuid)
+    val drug2 = testData.protocolDrug(protocolUuid = protocol1.uuid)
+    database.protocolDrugDao().save(listOf(drug1, drug2))
+
+    val drugsForProtocol2 = protocolRepository.drugsForProtocol(protocol2.uuid).blockingFirst()
+    assertThat(drugsForProtocol2).isEqualTo(protocolRepository.defaultProtocolDrugs())
+  }
+
+  @Test
+  fun when_protocols_are_present_in_database_then_they_should_be_returned() {
+    val config = configProvider.blockingGet()
+
+    if (config.isProtocolDrugSyncEnabled) {
```

Comment:
Prefer using early returns to avoid indenting a large code block. This can be changed to,

```kotlin
if (config.isProtocolDrugSyncEnabled.not()) {
  return
}
```